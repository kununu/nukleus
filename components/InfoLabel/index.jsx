import React, {PropTypes} from 'react';

import {
  controlLabelRequired,
  controlNote
} from '../index.scss';

const InfoLabel = ({requiredLabel, maxLength, displayLength, inputValue}) => {
  if (requiredLabel && (!displayLength || inputValue.trim() === '')) {
    return (<span className={`${controlNote} ${controlLabelRequired}`}>{requiredLabel}</span>);
  }

  // Show requiredLabel if available and user hasn't typed any inputValues
  return displayLength ? (
    <span className={`${controlNote} ${controlLabelRequired}`}>
      <strong>{inputValue.trim().length}</strong>/{maxLength}
    </span>
  ) : null;
};

InfoLabel.propTypes = {
  displayLength: PropTypes.bool,
  inputValue: PropTypes.string,
  maxLength: PropTypes.number.isRequired,
  requiredLabel: PropTypes.string
};

InfoLabel.defaultProps = {
  displayLength: false,
  inputValue: '',
  maxLength: 500,
  requiredLabel: ''
};

export default InfoLabel;
