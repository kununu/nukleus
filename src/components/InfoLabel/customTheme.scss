.infoLabel {
  font-size: 16px;
  color: blue;
  font-style: normal;
  text-align: right;
  font-weight: bold;
}