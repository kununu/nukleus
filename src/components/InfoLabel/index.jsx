import React from 'react';
import PropTypes from 'prop-types';

import {
  controlLabelRequired,
  controlNote,
} from '../index.scss';

const InfoLabel = ({
  displayLength,
  inputValue,
  maxLength,
  requiredLabel,
}) => {
  if (requiredLabel && (!displayLength || inputValue.trim() === '')) {
    return (<span className={`${controlNote} ${controlLabelRequired}`}>{requiredLabel}</span>);
  }

  // Show requiredLabel if available and user hasn't typed any inputValues
  return displayLength ? (
    <span className={`${controlNote} ${controlLabelRequired}`}>
      <strong>{inputValue.length}</strong>
/
      {maxLength}
    </span>
  ) : null;
};

InfoLabel.propTypes = {
  displayLength: PropTypes.bool,
  inputValue: PropTypes.string,
  maxLength: PropTypes.number,
  requiredLabel: PropTypes.string,
};

InfoLabel.defaultProps = {
  displayLength: false,
  inputValue: '',
  maxLength: 500,
  requiredLabel: '',
};

export default InfoLabel;
