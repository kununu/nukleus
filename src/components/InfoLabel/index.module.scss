@import '~@kununu/kununu-theme-v2/scss/base/variables';

.infoLabel {
  text-transform: uppercase;
  font-size: $font-size-xxs;
  color: $gray-base-60;
  font-style: normal;
  text-align: right;
  font-weight: $font-weight-light;
}