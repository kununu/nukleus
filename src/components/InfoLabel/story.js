import React from 'react';
import {storiesOf} from '@storybook/react';
import {defaultOptions} from 'defaultOptions'; // eslint-disable-line import/no-unresolved
import {withOptions} from '@storybook/addon-options';
import {
  withKnobs, text, boolean, number,
} from '@storybook/addon-knobs';
import {withInfo} from '@storybook/addon-info';

import ThemeProvider from '../ThemeProvider';

import customTheme from './customTheme.scss';

import InfoLabel from './index';

const textInfo = 'The info label component! This tiny component is used to show additional info about a form field to the user. It can either show a required text or the character count. You can select this based on the displayLength true/false or with a requiredLabel string.';

storiesOf('InfoLabel', module)
  .addDecorator(withOptions({...defaultOptions}))
  .addDecorator(withKnobs)
  .addDecorator((story, context) => withInfo(textInfo)(story)(context))
  .add('basics', () => (
    <div style={{maxWidth: '1000px'}}>
      <h3>InfoLabel</h3>
      <p>
        {textInfo}
      </p>
      <div style={{
        maxWidth: '100px',
        position: 'relative',
      }}
      >
        <InfoLabel
          inputValue={text('inputValue', 'Simple info label')}
          displayLength={boolean('displayLength', true)}
          maxLength={number('maxLength', 500)}
          requiredLabel={text('requiredLabel', 'Required')}
        />
      </div>
    </div>
  ))
  .add('custom', () => (
    <div style={{maxWidth: '1000px'}}>
      <h3>InfoLabel with custom theme</h3>
      <div style={{
        maxWidth: '100px',
        position: 'relative',
      }}
      >
        <ThemeProvider theme={customTheme}>
          <InfoLabel
            inputValue={text('inputValue', 'Simple info label')}
            displayLength={boolean('displayLength', true)}
            maxLength={number('maxLength', 500)}
            requiredLabel={text('requiredLabel', 'Required')}
          />
        </ThemeProvider>
      </div>
    </div>
  ));
