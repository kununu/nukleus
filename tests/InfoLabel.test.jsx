import React from 'react';
import renderer from 'react-test-renderer';

import InfoLabel from '../components/InfoLabel';

test('Renders an InfoLabel without crashing', () => {
  const component = renderer.create(
    <InfoLabel requiredLabel="required" />
  );
  expect(component.toJSON()).toMatchSnapshot();
});

test('Renders an InfoLabel with a characterCounter', () => {
  const component = renderer.create(
    <InfoLabel displayLength />
  );

  expect(component.toJSON()).toMatchSnapshot();
});

test('Renders an InfoLabel with a characterCounter and displays correct count and maxLength', () => {
  const component = renderer.create(
    <InfoLabel displayLength inputValue="test" maxLength={150} />
  );

  expect(component.toJSON()).toMatchSnapshot();
});

test('Renders an InfoLabel with a characterCounter and displays the requiredLabel, when value isn\'t set', () => {
  const component = renderer.create(
    <InfoLabel displayLength requiredLabel="required" />
  );

  expect(component.toJSON()).toMatchSnapshot();
});
