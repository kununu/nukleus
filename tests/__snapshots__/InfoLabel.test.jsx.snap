exports[`test Renders an InfoLabel with a characterCounter 1`] = `
<span
  className="controlNote controlLabelRequired">
  <strong>
    0
  </strong>
  /
  500
</span>
`;

exports[`test Renders an InfoLabel with a characterCounter and displays correct count and maxLength 1`] = `
<span
  className="controlNote controlLabelRequired">
  <strong>
    4
  </strong>
  /
  150
</span>
`;

exports[`test Renders an InfoLabel with a characterCounter and displays the requiredLabel, when value isn't set 1`] = `
<span
  className="controlNote controlLabelRequired">
  required
</span>
`;

exports[`test Renders an InfoLabel without crashing 1`] = `
<span
  className="controlNote controlLabelRequired">
  required
</span>
`;
